packages/core/content-type-builder/admin/src/components/FormModal/attributes/types.js

```diff
@@ -132,14 +131,26 @@ const types = {
     return yup.object(shape);
   },
   enumeration: (usedAttributeNames, reservedNames) => {
-    const ENUM_REGEX = new RegExp('^[_A-Za-z][_0-9A-Za-z]*$');
+    /**
+     * For enumerations the least common denomiator is GraphQL, where
+     * values needs to match the secure name regex:
+     * GraphQL Spec https://spec.graphql.org/June2018/#sec-Names
+     *
+     * Therefore we need to make sure our users only use values, which
+     * can be returned by GraphQL, by checking the regressed values
+     * agains the GraphQL regex.
+     *
+     * TODO V5: check if we can avoid this coupling by moving this logic
+     * into the GraphQL plugin.
+     */
+    const GRAPHQL_ENUM_REGEX = new RegExp('^[_A-Za-z][_0-9A-Za-z]*$');
 
     const shape = {
       name: yup
         .string()
         .test(alreadyUsedAttributeNames(usedAttributeNames))
         .test(isNameAllowed(reservedNames))
-        .matches(ENUM_REGEX, errorsTrads.regex)
+        .matches(GRAPHQL_ENUM_REGEX, errorsTrads.regex)
```

Comment:
Oh it does - the ENUM regex was taken from the GraphQL spec, because GraphQL defines the minimum we can do with these values (every other API is more flexible). This logic should ideally live in the GraphQL plugin and not be coupled in core, but that would be a bigger refactoring. I've tried to explain that in the code comment ...

Reply:
Yes I know (I was the one that did it at that time :p), but here it is the attribute name no? Nothing to do with GraphQL

Reply:
We discussed the issue on discord: for now we keep it as it is. I've create a follow-up issue that should be adressed at some point: https://github.com/strapi/strapi/issues/13761